docs/cloud-workload-security/scripts/templates/agent_expressions.md

```diff
@@ -140,7 +140,7 @@ Examples:
 
 Examples:
 * `T=8` and `width=8` both are in *args_options* for the command `ls -T 8 --width=8`
-* `exec.args_options ~= [ “s=.*\’” ]` can be used to detect `sudoedit` was launched with `-s` argument and a command that ends with a `\`
+* `exec.args_options in [ ~"s=.*\'" ]` can be used to detect `sudoedit` was launched with `-s` argument and a command that ends with a `\`
```

Comment:
wait doesn't this mean that the command ends in '? The \ is just to escape the '? Testing now

Reply:
ah yeah, I think you found a third reason why this is wrong

Reply:
this should most likely be `\\`

Reply:
i believe this is trying to detect this [vulnerability](https://www.sudo.ws/security/advisories/unescape_overflow/) ([additional resource](https://thesecmaster.com/fix-the-new-sudo-vulnerability-cve-2021-3156/)): `sudoedit -s /`. HOWEVER, the actual MITRE CVE [post](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-3156) says "backslash" which is "\". In any case, my version of sudoedit doesn't even have -s as a valid flag anymore. Gonna ask the PDEs if they're familiar with this CVE and if it's supposed to be \ or /.

I'm trying with `exec.args_options in [ ~"e=.*\/" ]` with grep (idk other commands that have -s as a flag that can take args off the top of my head), and it's not working atm. will report back

Reply:
hmmm i'm not sure this works. I changed the functional test to 
```
{
			ID: "test_rule_args_options",
			//Expression: `exec.args_options in ["block-size=123"]`,
			Expression: `exec.args_options in [ ~"block-size=.*3" ]`,
},
```
at `pkg/security/tests/process_test.go`. and ran ` inv -e security-agent.functional-tests --testflags "-test.v -test.run TestProcessContext/args-options"` which fails.

The test is executing `ls --block-size 123`

Reply:
ah I know why, ~ is for pattern, and we are providing a regexp: `.*` is a regex

Reply:
can you try with `r"block-size.."` ?

Reply:
and `~"block-size=*3`

Reply:
this should be good now in the last PR

Reply:
so i guess for this doc, we can do `exec.args_options in [ r"s=.*\/" ]`, assuming the vulnerability is indeed `sudoedit -s /` and MITRE is the one with the typo. `sudoedit -s \` I'm pretty sure would just get us into a new line unless sudoedit does some wild parsing of `\`

Reply:
I'm pretty sure the vuln is with a real backlash `\`. Reading https://blog.qualys.com/vulnerabilities-threat-research/2021/01/26/cve-2021-3156-heap-based-buffer-overflow-in-sudo-baron-samedit it seems:
- `\` is the underlying vulnerability
- `/` allows to check if your system is vulnerable ?

Reply:
ahhhh yes you're def right. consider my premature approval fully maturely approved